iterating-over-data/iterating-over-dictionaries.md

```diff
@@ -0,0 +1,321 @@
+# Iterating Over Dictionaries
+
+## Learning Goals
+
+- Practice using for loops to iterate through a dictionary
+
+## Introduction
+
+The overall concept for iterating over dictionaries is the same as iterating over lists. We can use `for` loops to iterate over **each key-value pair in a dictionary.**
+
+However, because dictionaries are key-value pairs, there are a few differences to pay attention to.
+
+## `for` Loops for Keys and Values Need `.items()`
+
+To iterate over a dictionary, we can use the following syntax:
+
+```python
+for my_key, my_value in my_dict.items():
+    print(my_key, my_value)
+```
+
+| Piece of Code              | Notes                                                                                                                                                        |
+| -------------------------- | ------------------------------------------------------------------------------------------------------------------------------------------------------------ |
+| `for`                      | `for` is a reserved keyword in Python. Python recognizes `for` as the beginning of a `for` loop.                                                             |
+| `my_key`                   | **Replace this** part with a name that represents what each key is. This local variable represents the _key_ in each key-value pair, one at a time.          |
+| `my_value`                 | **Replace this** part with a name that represents what each value is. This local variable represents the _value_ in each key-value pair, one at a time.      |
+| `in`                       | `in` is a reserved keyword in Python. It separates the name of the `my_element` variable and the list we're iterating on.                                    |
+| `my_dict`                  | **Replace this** part with the desired list to iterate over. This can be a variable that holds a list, a list literal, or any expression that becomes a list |
+| `.items()`                 | `.items()` is a method that dictionaries have. It returns a view of the dictionary, or a version of the dictionary's key-value pairs that are iterable.      |
+| `:`                        | This colon begins the `for` loop's body                                                                                                                      |
+| ` print(my_key, my_value)` | **Replace this** with any code that should execute during each loop. This is the loop's body.                                                                |
+
+### About `.items()`
+
+The most unique part about iterating over dictionaries is that we should call `.items()` next to each dictionary when starting the for loop.
+
+Python is unable to iterate over dictionaries as is; `.items()` lets us use the `for` loop with it. If we neglect to add `.items()`, we'll likely get this error:
+
+```bash
+ValueError: too many values to unpack (expected 2)
+```
```

Comment:
I'd like to go without including this for C15

And, I have NOT figured out how to make suggestions compatible with code blocks within them 🔥 😡 

I can't recreate your `ValueError` message; I'm going to stick with the original message 1. out of momentum 2. they're hopefully conceptually congruent enough (<- I made up that phrase) that they can be bridged, and I'd rather a student see the less-detailed one and accept the more detailed one as possible, rather than the other way around

Reply:
Weird... Out of curiosity, what code were you running?  I did 
```python
my_dict = { 'a': 0, 'b': 1}
for k, v in my_dict:
    print(k, v)
```

And to stick with the shorter content, I'd still make some suggestions, specifically targeted at focusing the material on the iteration described here being the key-value pairs. For instance, Python _is_ able to iterate over dictionaries without .items(), it just iterates over keys.  Maybe something like this?

==================

The most unique part about iterating over dictionary key-value pairs is that we must use `.items()` with the dictionary when starting the `for` loop. Notice that it appears right after the dictionary variable, with no space in between.

This syntax is required because Python doesn't iterate over the _items_ of a dictionary by default. `.items()` informs the dictionary that we'd like to use the key-value pairs in the `for` loop instead of the dictionary's default behavior. If we neglect to add `.items()`, we'll likely get this error:

```bash
ValueError: too many values to unpack (expected 2)
```